static/build/.tmp_update/script/miyoogamelist_gen.sh

```diff
@@ -0,0 +1,53 @@
+#!/bin/sh
+
+rootdir="/mnt/SDCARD/Roms"
+out='miyoogamelist.xml'
+
+generate_miyoogamelist() {
+    # create backup of previous miyoogamelist.xml
+    if [ -f "$out" ]; then
+        mv "$out" "$out.bak"
+    fi
+
+    echo '<?xml version="1.0"?>' >$out
+    echo '<gameList>' >>$out
+
+    for rom in *; do
+        # ignore subfolders because miyoogamelist don't work with them
+        # also ignores Imgs folder, nice
+        if [ -d "$rom" ]; then
+            continue
+        fi
+
+        # .xml, .bak & .db files are no games
+        # .bin files are no games, .cue files are!
+        if echo "$rom" | grep -qE "\.(xml|bak|db|bin)$"; then
+            continue
+        fi
+
+        filename="${rom%.*}"
+        digest=$(echo "$rom" | grep -oE "^([^(]*)" | awk '{$1=$1};1')
+
+        cat <<EOF >>$out
+    <game>
+        <path>./$rom</path>
+        <name>$digest</name>
+        <image>./Imgs/$filename.png</image>
+    </game>
+EOF
+    done
+
+    echo '</gameList>' >>$out
+}
+
+for system in "$rootdir"/*; do
+    if [ -d "$system" ]; then
+        # ignore Arcades & Ports since they have their own gamelist already
+        if echo "$system" | grep -qE "(PORTS|FBA2012|FBNEO|MAME2000|ARCADE|NEOGEO|CPS1|CPS1|CPS3)"; then
```

Comment:
Couldn't we clean the names inside an existing gamelist if applicable?

Reply:
I don't know about that yet. I am not a huge fan of this condition either but I am not sure how to handle the arcade roms naming in a nice way for now